### ISSUE TITLE
crash after deleting a torrent

### ISSUE THREAD

**Issue by sergserg2** [transmission/transmission#3152]
I've had it several times. After deleting the downloaded torrent (using Transmission Remote GUI), the entire application crashes (when is actively shared this torrent I think).

```
[2022-05-28 13:02:15.809] inf xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx Removing torrent (/home/software/transmission/transmission/libtransmission/torrent.cc:1603)
[2022-05-28 13:02:15.835] inf xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx Pausing torrent (/home/software/transmission/transmission/libtransmission/torrent.cc:1554)

```

Is it correct that removing a torrent is before pausing? Shouldn't it be reversed (pausing, so no shared, then removing)?

journalctl:
```
May 28 13:02:35 OdroidN2 systemd[1]: transmission-daemon.service: Main process exited, code=killed, status=11/SEGV
May 28 13:02:35 OdroidN2 systemd[1]: transmission-daemon.service: Failed with result 'signal'.
May 28 13:02:35 OdroidN2 systemd[1]: transmission-daemon.service: Consumed 1h 2min 21.200s CPU time.
```

**Comment by ckerr** [comment]
@sergserg2 what build of transmission-daemon are you using?

**Comment by sergserg2** [comment]
@ckerr build 690cf50e53 from github, problem was also on earlier ones

**Comment by ckerr** [comment]
@sergserg2  if you're building from source from github, any chance you could run `transmission-daemon -f` in a debugger and get a backtrace?

```

$ gdb transmission-daemon
(gdb) handle SIGPIPE nostop noprint nopass
(gdb) r -f
... run until Transmission crashes...
(gdb) thread apply all bt full
```

**Comment by ckerr** [comment]
I'm about 80% sure this is fixed by #3182. @sergserg2 please reopen this ticket or file a new one if this is incorrect
